online-dashboard-api/app/Http/Responses/ApiResponse.php

```diff
@@ -0,0 +1,35 @@
+<?php
```

Comment:
Excellent work! 🔥

Can we write the entire file in this manner! 

`public function setMessage($message): self
    {
        $this->message = $message;

        return $this;
    }`
    
    In place of getTokens we can replace with 
    `    public function mergeResults(array $results = []): self
    {
        $this->response = array_merge($this->results, $results);
        
        return $this;
    }`
    
    
For responses you can write the different function based on the code
like
`    public function respondOK(): JsonResponse
    {
        return $this->setStatusCode(200)->respondWithSuccess();
    }`
    
    
and use them whereever you want!